shared/profile/edit-profile/index.js

```diff
@@ -1,28 +1,59 @@
 // @flow
 import React, {Component} from 'react'
 import Render from './render'
+import {compose, lifecycle, withHandlers, withState} from 'recompose'
 import {connect} from 'react-redux'
 import {editProfile} from '../../actions/profile'
 import {maxProfileBioChars} from '../../constants/profile'
 import {navigateUp} from '../../actions/route-tree'
 
 import type {Props} from './render'
 
-class EditProfile extends Component<void, Props, void> {
+type State = {
+  bio: ?string,
+  fullname: ?string,
+  location: ?string,
+}
+
+const _bioLengthLeft = (bio: ?string) => {
+  return bio ? maxProfileBioChars - bio.length : maxProfileBioChars
+}
+
+const RenderWrapped = compose(
+  withState('bio', 'onBioChange', props => props.bio),
+  withState('fullname', 'onFullnameChange', props => props.fullname),
+  withState('location', 'onLocationChange', props => props.location),
+  withHandlers({
+    onSubmit: ({bio, fullname, location, onSubmit}) => () => onSubmit({bio, fullname, location}),
+  }),
+  lifecycle({
+    componentWillReceiveProps: function(nextProps) {
+      const bioLengthLeft = _bioLengthLeft(nextProps.bio)
+      this.setState({bioLengthLeft})
+    },
+  })
+)(Render)
+
+class EditProfile extends Component<void, Props, State> {
```

Comment:
this doesn't need state, it comes in as a prop using withState. therefore it can be stateless